lib/private/Lock/DBLockingProvider.php

```diff
@@ -192,10 +146,13 @@ public function acquireLock(string $path, int $type, string $readablePath = null
 			}
 			$result = $this->initLockField($path, -1);
 			if ($result <= 0) {
-				$result = $this->connection->executeUpdate(
-					'UPDATE `*PREFIX*file_locks` SET `lock` = -1, `ttl` = ? WHERE `key` = ? AND `lock` = ?',
-					[$expire, $path, $existing]
-				);
+				$query = $this->connection->getQueryBuilder();
+				$query->update('file_locks')
+					->set('lock', $query->createNamedParameter(-1))
```

Comment:
Not an expert but, isn't the type mandatory if it's not a string? You actually used it below: https://github.com/nextcloud/server/pull/32077/files#diff-8d8c58b043af460a222699907706f7cd2529c1ce1524b1e67b986d47e403ed78R172

```suggestion
					->set('lock', $query->createNamedParameter(-1), IQueryBuilder::PARAM_INT));
```

Reply:
yeah I'm also a bit confused by that, some old codes in this class also didn't mention the type for int. But yeah I think it's a good idea to add the types as it can't hurt

Reply:
If you add it here, then there are a few other places in your changes that could use it :)